intelij-plugin/src/main/highlighter/tech/pantheon/yanginator/plugin/YangHighlighter.bnf

```diff
@@ -0,0 +1,222 @@
+      H_ERROR_MESSAGE_KEYWORD = 'error-message'
+      H_EXTENSION_KEYWORD   = 'extension'
+      H_DEVIATION_KEYWORD   = 'deviation'
+      H_DEVIATE_KEYWORD     = 'deviate'
+      H_FEATURE_KEYWORD     = 'feature'
+      H_FRACTION_DIGITS_KEYWORD = 'fraction-digits'
+      H_GROUPING_KEYWORD    = 'grouping'
+      H_IDENTITY_KEYWORD    = 'identity'
+      H_IF_FEATURE_KEYWORD  = 'if-feature'
+      H_IMPORT_KEYWORD      = 'import'
+      H_INCLUDE_KEYWORD     = 'include'
+      H_INPUT_KEYWORD       = 'input'
+      H_KEY_KEYWORD         = 'key'
+      H_LEAF_KEYWORD        = 'leaf'
+      H_LEAF_LIST_KEYWORD   = 'leaf-list'
+      H_LENGTH_KEYWORD      = 'length'
+      H_LIST_KEYWORD        = 'list'
+      H_MANDATORY_KEYWORD   = 'mandatory'
+      H_MAX_ELEMENTS_KEYWORD = 'max-elements'
+      H_MIN_ELEMENTS_KEYWORD = 'min-elements'
+      H_MODULE_KEYWORD      = 'module'
+      H_MUST_KEYWORD        = 'must'
+      H_NAMESPACE_KEYWORD   = 'namespace'
+      H_NOTIFICATION_KEYWORD= 'notification'
+      H_ORDERED_BY_KEYWORD  = 'ordered-by'
+      H_ORGANIZATION_KEYWORD= 'organization'
+
+      H_OUTPUT_KEYWORD      = 'output'
+      H_PATH_KEYWORD        = 'path'
+      H_PATTERN_KEYWORD     = 'pattern'
+      H_POSITION_KEYWORD    = 'position'
+      H_PREFIX_KEYWORD      = 'prefix'
+      H_PRESENCE_KEYWORD    = 'presence'
+      H_RANGE_KEYWORD       = 'range'
+      H_REFERENCE_KEYWORD   = 'reference'
+      H_REFINE_KEYWORD      = 'refine'
+      H_REQUIRE_INSTANCE_KEYWORD = 'require-instance'
+      H_REVISION_KEYWORD    = 'revision'
+      H_REVISION_DATE_KEYWORD = 'revision-date'
+      H_RPC_KEYWORD         = 'rpc'
+      H_STATUS_KEYWORD      = 'status'
+      H_SUBMODULE_KEYWORD   = 'submodule'
+      H_TYPE_KEYWORD        = 'type'
+      H_TYPEDEF_KEYWORD     = 'typedef'
+      H_UNIQUE_KEYWORD      = 'unique'
+      H_UNITS_KEYWORD       = 'units'
+      H_USES_KEYWORD        = 'uses'
+      H_VALUE_KEYWORD       = 'value'
+      H_WHEN_KEYWORD        = 'when'
+      H_YANG_VERSION_KEYWORD= 'yang-version'
+      H_YIN_ELEMENT_KEYWORD = 'yin-element'
+
+      // other keywords
+      H_CURRENT_KEYWORD     = 'current'
+      H_DELETE_KEYWORD      = 'delete'
+      H_DEPRECATED_KEYWORD  = 'deprecated'
+      H_FALSE_KEYWORD       = 'false'
+      H_ADD_KEYWORD         = 'add'
+      H_MAX_KEYWORD         = 'max'
+      H_MIN_KEYWORD         = 'min'
+      H_NOT_SUPPORTED_KEYWORD = 'not-supported'
+      H_OBSOLETE_KEYWORD    = 'obsolete'
+      H_REPLACE_KEYWORD     = 'replace'
+      H_SYSTEM_KEYWORD      = 'system'
+      H_TRUE_KEYWORD        = 'true'
+      H_UNBOUNDED_KEYWORD   = 'unbounded'
+      H_USER_KEYWORD        = 'user'
+
+      H_WHITE_SPACE = 'regexp:\s+'
+      H_STRING = "regexp:\"([^\"])*\""
+      H_DATE_ARG='regexp:[0-9]{4}-[0-9]{1,2}-[0-9]{1,2}'
+      H_IDENTIFIER = 'regexp:[a-zA-Z_][0-9a-zA-Z_\-.]*'
+      H_EXTENSION_IDENTIFIER = 'regexp:[a-zA-Z_][0-9a-zA-Z_\-.:]*'
```

Comment:
Does this allow for "extension foo:bar:baz"?

Reply:
this is highlighter's grammar this does